docarray/array/stacked/array_stacked.py

```diff
@@ -278,84 +236,65 @@ def __getitem__(self: T, item: int) -> T_doc:
     def __getitem__(self: T, item: IndexIterType) -> T:
         ...
 
-    def __getitem__(self, item):
+    def __getitem__(self: T, item: Union[int, IndexIterType]) -> Union[T_doc, T]:
         if item is None:
             return self  # PyTorch behaviour
         # multiple docs case
         if isinstance(item, (slice, Iterable)):
-            item_ = cast(Iterable, item)
-            return self._get_from_data_and_columns(item_)
+            return self.__class__.from_columns_storage(self._storage[item])
         # single doc case
-        doc = self._docs[item]
-        return doc
-
-    @overload
-    def __setitem__(self: T, key: int, value: T_doc):
-        ...
+        return self.document_type.from_view(ColumnStorageView(item, self._storage))
 
-    @overload
-    def __setitem__(self: T, key: IndexIterType, value: T):
-        ...
+    def _get_array_attribute(
+        self: T,
+        field: str,
+    ) -> Union[List, 'DocumentArrayStacked', AbstractTensor]:
+        """Return all values of the fields from all docs this array contains
```

Comment:
I'm not sure I undrstand this sentence. Isn't it equivalend to "return one column of the data"? If so, I would also rename the method to that effect